interfaces/builtin/serial_port.go

```diff
@@ -208,6 +209,34 @@ func (iface *serialPortInterface) AutoConnect(*snap.PlugInfo, *snap.SlotInfo) bo
 	return true
 }
 
+func (iface *serialPortInterface) HotplugDeviceDetected(di *hotplug.HotplugDeviceInfo) (*hotplug.ProposedSlot, error) {
+	bus, _ := di.Attribute("ID_BUS")
+	if di.Subsystem() != "tty" || bus != "usb" || !serialDeviceNodePattern.MatchString(di.DeviceName()) {
+		return nil, nil
+	}
+
+	slot := hotplug.ProposedSlot{
+		Attrs: map[string]interface{}{
+			"path": di.DeviceName(),
+		},
+	}
+	if vendor, ok := di.Attribute("ID_VENDOR_ID"); ok {
+		slot.Attrs["usb-vendor"] = vendor
+	}
+	if product, ok := di.Attribute("ID_MODEL_ID"); ok {
+		slot.Attrs["usb-product"] = product
+	}
+	return &slot, nil
+}
+
+func (iface *serialPortInterface) HandledByGadget(di *hotplug.HotplugDeviceInfo, slot *snap.SlotInfo) bool {
+	var path string
+	if err := slot.Attr("path", &path); err != nil {
```

Comment:
the path can be either  the path of the device, or path of the symlink to create for it. That is the case if there are other usb-*  attributes so if possible we need to match on those first if present (see hasUsbAttrs helper)

Reply:
Done.